extensions/sql-migration/src/dialog/skuRecommendationResults/GenerateProvisioningScriptDialog.ts

```diff
@@ -0,0 +1,223 @@
+						case utils.MigrationTargetType.SQLMI:
+							destinationFilePath = path.join(folder, 'ARMTemplate-AzureSqlManagedInstance' + new Date().toISOString().split('T')[0] + '.json');
+							break;
+
+						case utils.MigrationTargetType.SQLVM:
+							destinationFilePath = path.join(folder, 'ARMTemplate-AzureSqlVirtualMachine' + new Date().toISOString().split('T')[0] + '.json');
+							break;
+
+						case utils.MigrationTargetType.SQLDB:
+							destinationFilePath = path.join(folder, 'ARMTemplate-AzureSqlDatabase' + new Date().toISOString().split('T')[0] + '.json');
+							break;
+					}
+					fs.writeFileSync(destinationFilePath!, this._armTemplateText);
+				} else {
+					console.log(this.model._armTemplateResult.generateTemplateError);
+					// log in telemetry -- TO DO
+				}
+			}
+		});
+
+		const uploadTemlateToAzureButton = _view.modelBuilder.button()
+			.withProps({
+				buttonType: azdata.ButtonType.Normal,
+				label: constants.UPLOAD_TEMPLATE_TO_AZURE,
+				width: 127,
+				height: 36,
+				iconHeight: 16,
+				iconWidth: 16,
+				iconPath: IconPathHelper.import,
+				CSSStyles: {
+					...styles.TOOLBAR_CSS
+				}
+			}).component();
+
+		uploadTemlateToAzureButton.onDidClick(async () => {
+			const selectAzureAccountDialog = new SelectStorageAccountDialog(this.model, this._targetType);
+			await selectAzureAccountDialog.initialize();
+		});
+
+		const buttonsContainer = _view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'display': 'flex',
+				'flex-direction': 'row',
+				'margin': '5px 5px 5px 5px',
+				'width': '244px',
+				'height': '36px',
+			}
+		}).component();
+
+		buttonsContainer.addItem(saveTemplateButton);
+		buttonsContainer.addItem(uploadTemlateToAzureButton);
+
+		const container = _view.modelBuilder.flexContainer().
+			withProps({
+				CSSStyles: {
+					'margin': '10px 10px 10px 10px',
+					'flex-direction': 'column',
+				}
+			})
+			.withItems([
+				armTemplateDescription,
+				buttonsContainer,
+				textContainer,
+			]).component();
+
+		return container;
+	}
+
+	private async displayArmTemplate(): Promise<void> {
+		this._armTemplateText = (this.model._armTemplateResult.template)!;
+		this._armTemplateTextBox.value = this._armTemplateText;
+	}
+
+
+
+	public async openDialog() {
+		if (!this._isOpen) {
+			this._isOpen = true;
+
+			this.dialog = azdata.window.createModelViewDialog(constants.TARGET_PROVISIONING_TITLE, 'ViewArmTemplateDialog', 'medium');
+
+			this.dialog.okButton.label = GenerateProvisioningScriptDialog.CloseButtonText;
+			this.dialog.okButton.position = 'left';
+			this._disposables.push(this.dialog.okButton.onClick(async () => await this.execute()));
+			this.dialog.cancelButton.hidden = true;
+
+			const dialogSetupPromises: Thenable<void>[] = [];
+			dialogSetupPromises.push(this.initializeDialog(this.dialog));
+
+			azdata.window.openDialog(this.dialog);
+			await Promise.all(dialogSetupPromises);
+
+			await this.model.getArmTemplate(this._targetType);
```

Comment:
this must be under try catch block. and log error if it fails.

Reply:
there is a try catch block in the getArmTemplate function and error is also logged .